### ISSUE TITLE
failing gradient verification on PT 1.8 with GPU

### ISSUE THREAD

**Issue by Borda** [PyTorchLightning/lightning-bolts#589]
## 🐛 Bug

Failing verification with Out of memory

```
t = Parameter containing:
tensor([[[[ 0.0232, -0.0209,  0.0345],
          [ 0.0014, -0.0473, -0.0385],
          [ 0.0365...-0.0871, -0.0187],
          [-0.0882,  0.0994, -0.0005],
          [-0.0846, -0.0062,  0.0862]]]], requires_grad=True)

    def convert(t):
        if convert_to_format is not None and t.dim() == 4:
            return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None,
                        non_blocking, memory_format=convert_to_format)
>       return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
```

### To Reproduce

https://dev.azure.com/PytorchLightning/lightning%20Bolts/_build/results?buildId=2931&view=logs&j=3afc50db-e620-5b81-6016-870a6976ad29&t=98354d77-e326-51ec-536f-1549451db1fa&l=1984
